### ISSUE TITLE
Allow Zero width no-break space in float parser

### ISSUE THREAD

**Issue by jonashaag** [pola-rs/polars#16721]
### Description

Found this character in a CSV file.

```
>>> pl.Series(["-\ufeff42"]).cast(float)
ComputeError: conversion from `str` to `f64` failed in column '' for 1 out of 1 values: ["-42"]
```

**Comment by stinodego** [comment]
Why do you think this should be convertible to a float? Looks like a bad input to me.

**Comment by jonashaag** [comment]
I found this in a random CSV file that I wanted to parse. I guess it's a relatively rare but legitimate ("non-")character that's inserted by Excel/etc for number formatting

**Comment by mcrumiller** [comment]
Was it the first two bytes? `FEFF` is the optional [byte-order mark](https://en.wikipedia.org/wiki/Byte_order_mark) for BE Utf-16, and it's probably ending up in your CSV due to being exported with a specified encoding and byte order. I've had plenty of CSVs that get wonky due to this mark, and I usually open them and re-encode them as UTF-8 in Notepad++ or something similar.

I don't know if polars checks the byte order mark if it exists. Maybe it should, but if so that should be its own issue. But it doesn't seem to have caused many problems for people up until now.

**Comment by jonashaag** [comment]
No, it was between two separators like this

```
transaction_id;amount
abcdef;-\ufeff42
```

**Comment by mcrumiller** [comment]
According to https://www.unicode.org/faq/utf_bom.html#bom6, that shouldn't be there at all. I'd do a replace prior to trying to do anything with that string.

**Comment by jonashaag** [comment]
Hmm ok sounds like an actual bug in my data then.

The only annoying thing that remains a potential todo is to improve displaying of the character in the error message. Although not sure how to display it since it's a "zero width" character...

**Comment by stinodego** [comment]
Yeah this doesn't sound like something Polars should fix. Closing then!
